src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/StorePal.macOS.LoaderPal.cs

```diff
@@ -119,6 +119,11 @@ public void MoveTo(X509Certificate2Collection collection)
 
                         X509Certificate2 cert = new X509Certificate2(newPal);
                         collection.Add(cert);
+
+                        if (newPal != pal)
+                        {
+                            pal.Dispose();
+                        }
```

Comment:
Note: With this object disposed in deterministic manner the existing tests should reliably trip on miscounted references without a GC being triggered.